File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    answer = ''
    temp = polynomial.split(" ")
    x_value = 0
    num = 0
    for i in temp:
        if i[0] == '+': continue
        if i[-1] == 'x':
            if i[0] == 'x':
                x_value += 1
            else :
                xx = i[:-1]
                x_value += int(xx)
        else :
            num += int(i)
    
    if x_value > 0:
        if x_value == 1:
            answer += 'x'
        else:
            answer += str(x_value) + 'x'
    
    if num > 0:
        if len(answer) != 0:
            answer += ' + '
        answer += str(num)
    return answer

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.cpp
#include <string>
#include <vector>
#include <sstream>

using namespace std;

string solution(string polynomial) {
    stringstream ss(polynomial);
    string temp = "";
    int num=0;
    int x=0;
    while(ss >> temp){
        if (temp == "+") continue;
        if(temp.back() == 'x'){
            if(temp == "x"){
                x += 1;
            }
            else{
                x += stoi(temp.substr(0, temp.size() - 1));
            }
        }
        else {
            num += stoi(temp);
        }
    }
    
    string answer = "";
    if (x > 0){
        if(x == 1){
            answer += "x";
        }
        else{
            answer += to_string(x) + "x";
        }
    }
    
    if (num > 0){
        if (!answer.empty()){
            answer += " + ";
        }
        answer += to_string(num);
    }
    
    return answer;
}